tensorflow/python/keras/metrics.py

```diff
@@ -769,8 +769,11 @@ def __init__(self, name='categorical_accuracy', dtype=None):
 @keras_export('keras.metrics.SparseCategoricalAccuracy')
 class SparseCategoricalAccuracy(MeanMetricWrapper):
   """Calculates how often predictions matches integer labels.
-
-  You can provide logits of classes as `y_pred`, since argmax of
+  
+  For example, if `y_true` is `[[2], [1]]` and `y_pred` is
```

Comment:
This change can be omitted since the example below conveys the same information.

Reply:
The issue I referenced asks for better documentation explaining why values are calculated as such. Do you still recommend that I omit the explanation?

Reply:
The sentence here is just putting the below example into words. We are not really explaining how we got the values here ie the formula used etc. This explanation is something I have been meaning to add to metrics soon. We have added such documentation to losses.py already. Please take a look.

Reply:
Thanks for the feedback. I looked at `losses.py` and understood what you meant. I omitted the textual explanation and wrote an equation in lieu of it, but I'm not sure if this is consistent with the style of the equations in `losses.py`. In particular, I assumed that `==` would be interpreted as an element-wise boolean operator, which is not the case in default Python syntax, nor is it a coherent mathematical expression. Do you have any pointers or suggestions? Thank you.

Reply:
You can use  np.equal in place of that. Also weighting denom should be sum of sample_weights, not n_pred. Thank you!

Reply:
Thanks for the pointer. I was originally thrown off by the note that `sample_weight` defaults to 1, which is why I thought division by `n_pred` is necessary (I mistakenly thought that the weights would default to something like `[1, 1]`, which isn't true).

However, after some more thought, I'm wondering if we need a weighting denominator at all, given that the sum of `sample_weight` is always going to be 1. In other words, would `acc = np.dot(sample_weight, np.equal(y_true, np.argmax(y_pred, axis=1))` suffice? Thanks again!

Reply:
@pavithrasv I pushed a commit based on your previous feedback. If time permits, I'd appreciate your review! Thanks in advance.